src/include/messenger/messenger.h

```diff
@@ -293,6 +322,8 @@ class Messenger : public common::DedicatedThreadTask {
    */
   void ProcessMessage(const ZmqMessage &msg);
 
+  bool UpdateMessagesSeen(const std::string &replica, message_id_t message_id);
```

Comment:
I think this would benefit from a docstring, just something quick explaining that we keep track of the max message id seen and then every unseen message id below it.

Reply:
added!